frame/support/src/traits/tokens/pay.rs

```diff
@@ -99,3 +99,29 @@ impl<A: TypedGet, F: fungible::Mutate<A::Type>> Pay for PayFromAccount<F, A> {
 	#[cfg(feature = "runtime-benchmarks")]
 	fn ensure_concluded(_: Self::Id) {}
 }
+
+pub struct PayFungibles<F, A>(sp_std::marker::PhantomData<(F, A)>);
+impl<A: TypedGet, F: fungibles::Mutate<A::Type> + fungibles::Inspect<A::Type>> Pay
+	for PayFungibles<F, A>
+{
+	type Balance = F::Balance;
+	type Beneficiary = A::Type;
+	type AssetKind = F::AssetId;
+	type Id = ();
+	fn pay(
+		who: &Self::Beneficiary,
+		asset_id: Self::AssetKind,
+		amount: Self::Balance,
+	) -> Result<Self::Id, ()> {
+		<F as fungibles::Mutate<_>>::transfer(asset_id, &A::get(), who, amount, Expendable)
+			.map_err(|_| ())?;
+		Ok(())
+	}
+	fn check_payment(_: ()) -> PaymentStatus {
+		PaymentStatus::Success
+	}
+	#[cfg(feature = "runtime-benchmarks")]
+	fn ensure_successful(_: &Self::Beneficiary, _: Self::Balance) {}
```

Comment:
Yeah, that's true. But I would have to do that in a separate PR, because the type signature of the trait methods will have to change to accept an asset_id. `fungibles::Mutate::mint` will require it